### PR TITLE
Several nes_vt Fixes and Improvements

### PR DESCRIPTION
**Pull request by daveshah1**
- Add support for a newer VTxx chipset with scrambled instructions and RGB444
  palette, used in the FC Pocket and DGUN-2573. This includes adding a new
	CPU type for this, which is a 6502 with a "scrambled" mode where the opcodes
	are XORed with 0xA1.
	
- Add the following new systems (dumped by myself unless credit specified):
		* SY-888B (similar to SY-889)
		* 89 in 1 Mini Game Console [TeamEurope]
		* CoolBoy RS-18 280 in 1 [TeamEurope]
		* Pocket Games 150 in 1 [taken from NES software list]
		* Handheld 210 in 1 [taken from NES software list]
	  * DV-Tech Nimbus [cah3e3]
		* CoolBoy RS-8 168 in 1 [TeamEurope]
		* 200 in 1 Retro TV Game
		* FC Pocket 600 in 1
		* MOGIS M320
		* Family Pocket 638 in 1
	  * Hummer Z-Dog [unknown]
	
- Add support for scrambled banking, so the Sports Game 69 in 1, Samuri 60 in 1,
  Z-Dog, and Power Joy now work correctly.

- General Improvements/Fixes including:
		* Improve PPU colour palette (credit to NewRisingSun for reference code).
		* Fix scanline IRQs, stops Super Mario Bros 3 and Z-Dog from crashing
		* Fix PPU data reads, fixes Super Mario Bros title screen and a few other
			games
 		* Other small accuracy improvements

### CODE REVIEW

**Comment by mmicko** [comment]
@galibert can you please confirm if 6502 additions are fine. All looks clean, but another pair of eyes always welcome.

File: src/devices/cpu/m6502/m6502.h [hunk 1]
```diff
@@ -136,8 +136,8 @@ class m6502_device : public cpu_device {
 	uint8_t read_arg(uint16_t adr) { return mintf->read_arg(adr); }
 	uint8_t read_pc() { return mintf->read_arg(PC++); }
 	uint8_t read_pc_noinc() { return mintf->read_arg(PC); }
-	void prefetch();
-	void prefetch_noirq();
+	virtual void prefetch();
```
Comment (galibert) [inline src/devices/cpu/m6502/m6502.h]: Don't do that.  Specialize the memory accessor instead and change read_sync/decrypt8.  See mame/machine/deco222.* for an example.

File: src/devices/cpu/m6502/m6502_vtscr.cpp [hunk 2]
```diff
@@ -0,0 +1,106 @@
+// license:BSD-3-Clause
+// copyright-holders:David Shah
+/***************************************************************************
+
+    m6502_vtscr.cpp
```
Comment (galibert) [inline src/devices/cpu/m6502/m6502_vtscr.cpp]: Put that in mame/machine/* or directly in nes_vt.* please.  No sane reason to have it in devices/cpu/m6502

**Comment by galibert** [comment]
Nah, it's very much not fine.  I've commented on the web interface, but I
may have fucked it up...

  OG.

**Comment by galibert** [comment]
In case it doesn't go through, the summary is "follow what is done in
src/mame/machine/deco222.*, and you'll get disassembly too".

  OG.

**Comment by daveshah1** [comment]
Thanks for your help, I think I understand now. Am I correct in saying that read_sync is only used for opcode fetches (as only opcodes are "encrypted").

**Comment by galibert** [comment]
Yup.

  OG.

**Comment by galibert** [comment]
Incidentally, it's called read_sync because the 6502 activates a signal
called sync (pin 7 on the original 6502) when reading an opcode byte.

  OG.

**Comment by daveshah1** [comment]
Thanks again for your help, how is this looking now after b0bbe2c?

**Comment by galibert** [comment]
Much better.  The debugger should show the decrypted code too now, right?

  OG.

**Comment by daveshah1** [comment]
Yes, disassembly is working correctly now in both modes.

**Pull request merged**